Cloud common - compare and contrast the three different service types (IaaS, PaaS, SaaS) closes 188 (#463)
bob-crutchley/courseware: topics/cloud/modules/iaas-paas-saas/README.md

# Difference between: IaaS, PaaS, SaaS

<!--TOC_START-->
## Contents
- [Overview](#overview)
	- [Tasks](#tasks)

<!--TOC_END-->
## Overview

> The main difference between the three models could be summed up to:

Reviewer: ```suggestion
The main differences between the three models can be summed up as follows:
```

- **IaaS** gives maximum flexibility for hosting custom built applications or acting as a data centre for data storage
- **PaaS** built on top of IaaS reduces the need for administering the system. 

> Freed up time from not needing to manage infrastructure can then be used for development of the application.

Reviewer: ```suggestion
Frees up time: by not needing to manage infrastructure, time can then be used for development of the application.
```

- **SaaS** is an out-of-the-box solution ready to be used that meets a specific business use.
Built on top of IaaS and PaaS.

Here's a diagram of the different responsibilities between each model:
![IaaS, PaaS, SaaS](https://imgur.com/3qRMo6r.jpg)

> When picking which model to use, think about how much control do you require for the given task, this will make it easier to decide which model to select.

Reviewer: ```suggestion
When picking which model to use, think about how much control you might require for the given task.

 This will make it easier to decide which model to select.
```
